plugins/modules/gitlab_group_variable.py

```diff
@@ -313,6 +322,8 @@ def native_python_main(this_gitlab, purge, requested_variables, state, module):
             item['environment_scope'] = '*'
         if item.get('variable_type') is None:
             item['variable_type'] = 'env_var'
+        if item.get('raw') is None:
+            item['raw'] = False
```

Comment:
> I think the concerns in the issue might be relevant - won't this add a raw key to each item in requested_variables even in GitLab 15.5 and earlier, resulting in the variable added here:
> ```py
> remove = [x for x in existing_variables if x not in requested_variables]
> ```
> because each item's var.attributes in existing_variables will not contain it?

I try to deploy a GitLab < 15.6 and made some tests against it later.

> Obviously this is also true for other keys added here. But maybe the remove variable should only depend on key and environment scope to be safe.

This is sufficient when state is `absent`. And it is implemented like that: https://github.com/ansible-collections/community.general/blob/main/plugins/modules/gitlab_project_variable.py#L366-L374  
But I couldn' remember why `masked` and `protected` properties aren't pop'd here too. 

However, when `purge: true` is requested, we really want to remove  _e v e r y_   variable.  
And the entire diff processing becomes completely undefined when the module doesn't know and handle the API defaults that the user maybe doesn't request.

E.g. when the user isn't request the `masked` property. How should the module behave?   
a) Use the APIs default? - That's the current behaviour
b) Keep the returned GitLab value? - That would result in a breaking change.

The only thing I can think of is, introduce a new parameter `skip_default_values_if_not_requested` (_or some more graceful name_) with default `false`. That would make `b)` possible without breaking backwards compatibility.   
But this would need more time to implement and to test.  
I feel like to hotfix this issue with just appending the `raw` property to fix this behaviour asap.

> Although, who knows what GitLab will do next 

Yes. That's a big issue and we were facing it since the modules exist.

Reply:
Another possibility is, when `purge: true` is requested, that first, the module removes all variables and after that, recreate them.  
That would also made the module immune to new properties that GitLab might introduce.
But the change detection will still get broken than.  

That's also more a hotfix and not that graceful ...  
What do you think @felixfontein

Reply:
That sounds pretty ugly.

How about 'simply' ignoring all unknown properties (that aren't explicitly listed in https://github.com/ansible-collections/community.general/blob/da417c956b25986854edd0f49f0330d8c22e6891/plugins/modules/gitlab_project_variable.py#L420-L425) during idempotence checks, and also during purge?